src/model/books/books.selectors.js

```diff
@@ -9,10 +10,26 @@ const addReadPropertyByISBN = (books, ISBN) => {
   return R.adjust(addReadPropToObj, index, books);
 };
 
+const fuzzy = books => (term) => {
+  const searcher = new FuzzySearch(books, ['ISBN', 'author', 'title'], {
+    caseSensitive: false,
+  });
+  return searcher.search(term);
+};
+
 export const getBooks = createSelector(
   [getRawBooks, getBooksRead],
   (rawBooks, ISBNsOfBooksIRead) => R.reduce(
     addReadPropertyByISBN,
     rawBooks,
     ISBNsOfBooksIRead,
   ));
+
+export const getFilteredBooks = createSelector(
+  [getBooks],
+  (books) => {
+    if (books.length > 0) {
+      return fuzzy(books)('Steve');
+    }
+    return books;
+  });
```

Comment:
The `createSelector` function uses a memoized pattern to speed up computations against changes in Redux. In other words if the value from redux doesn't change then the expression wont fire - it will return a cached version of the previous expression call. 

This `selector` could be updated to pull a search term of the redux store and replace 'steve'.